source/dagger/gameplay/team_game/golem.cpp

```diff
@@ -0,0 +1,180 @@
+			}
+			else if (golem_.currentAction == EAction::Moving) {
+				auto nextPosition = golem_.postition - 1;
+
+				auto coords = WalkingPath::path.back();
+				int i = 0;
+				for (auto iter : WalkingPath::path) {
+					if (i == nextPosition) {
+						coords = iter;
+						break;
+					}
+					i++;
+				}
+
+				auto destinationX = coords.x; // Maybe add +/- a few percent to make paths bit more varied
+				auto destinationY = coords.y;
+
+				if (transform_.position.x < destinationX) {
+					golem_.direction.x = 1;
+				}
+				else if (transform_.position.x > destinationX) {
+					golem_.direction.x = -1;
+				}
+				else golem_.direction.x = 0;
+				if (transform_.position.y < destinationY) {
+					golem_.direction.y = 1;
+				}
+				else if (transform_.position.y > destinationY) {
+					golem_.direction.y = -1;
+				}
+				else golem_.direction.y = 0;
+
+				if (golem_.direction.x != 0) {
+					sprite_.scale.x = (-golem_.direction.x) * abs(sprite_.scale.x);
+					AnimatorPlay(animation_, "ancient_defenders:golem:WALK_SIDE");
+				}
+				else if (golem_.direction.y == 1) {
+					AnimatorPlay(animation_, "ancient_defenders:golem:WALK_SIDE");
+				}
+				else {
+					AnimatorPlay(animation_, "ancient_defenders:golem:WALK_SIDE");
+				}
+
+				transform_.position.x += golem_.direction.x * golem_.speed * Engine::DeltaTime();
+				transform_.position.y += golem_.direction.y * golem_.speed * Engine::DeltaTime();
+
+				if ((transform_.position.x < destinationX && golem_.direction.x == -1) ||
+					(transform_.position.x > destinationX && golem_.direction.x == 1)) {
+					transform_.position.x = destinationX;
+				}
+				else if ((transform_.position.y > destinationY && golem_.direction.y == 1) ||
+					(transform_.position.y < destinationY && golem_.direction.y == -1))
+				{
+					transform_.position.y = destinationY;
+				}
+
+				if (transform_.position.x == coords.x && transform_.position.y == coords.y) {
+					golem_.postition--;
+					if ((golem_.postition - 1) < 0) {
+						golem_.currentAction = EAction::Idling;
+						return;
+					}
+				}
+
+			}
+			else if (golem_.currentAction == EAction::Attacking) {
+				if (golem_.direction.x != 0) {
+					sprite_.scale.x = (-golem_.direction.x) * abs(sprite_.scale.x);
+					AnimatorPlay(animation_, "ancient_defenders:golem:ATTACK_FRONT");
+				}
+				else if (golem_.direction.y == 1) {
+					AnimatorPlay(animation_, "ancient_defenders:golem:ATTACK_FRONT");
+				}
+				else {
+					AnimatorPlay(animation_, "ancient_defenders:golem:ATTACK_FRONT");
+				}
+				Engine::Registry().get<MageStats>(range_.target).health -= golem_.meleeDmg;
+				golem_.currentAction = EAction::Moving; // Go back to moving after attacking
+			}
+			else if (golem_.currentAction == EAction::Chanting) {
+				// Increase tower building progres
+			}
+			else if (golem_.currentAction == EAction::Defending) {
+				// Cast spells
+			}
+		});
+
+}
+
+void ancient_defenders::GolemBehaviorSystem::OnEndOfFrame()
+{
```

Comment:
We will move the responsibility of removing characters to HealthManagementSystem (still in works) so this should be removed both here and in mage.cpp.